File: Day_19/Q73.cpp
#include <bits/stdc++.h>
using namespace std;
//Input The 2D-Array
void input_array(vector<vector<int>> &arr,int rows,int col){
    for(int i = 0;i < rows;i++){
        for(int j = 0;j < col;j++) cin >> arr[i][j];
    }
}
//Display The 2D-Array
void display_matrices(vector<vector<int>> &arr,int rows,int col){
    for(int i = 0;i < rows;i++){
        for(int j = 0;j < col;j++) cout << arr[i][j] << "\t";
        cout << endl;
    }
}
//Adding The Matrices
vector<vector<int>> matrix_sum(vector<vector<int>> &matrix1,vector<vector<int>> &matrix2,int rows,int col){
    vector<vector<int>> summation(rows,vector<int>(col));
    for(int i = 0;i < rows;i++){
        for(int j = 0;j < col;j++) summation[i][j] = matrix1[i][j] + matrix2[i][j];
    }
    return summation;
}
int main(){
    int rows,col;
    cin >> rows >> col;
    //Input Validation
    if(rows <= 0 || col <= 0){
        cout << "Invalid Input";
        return 0;
    }
    vector<vector<int>> matrix1(rows,vector<int>(col));
    vector<vector<int>> matrix2(rows,vector<int>(col));
    //Performing Necessary Actions By Calling The Functions
    input_array(matrix1,rows,col);
    input_array(matrix2,rows,col);
    vector<vector<int>> summation_matrix = matrix_sum(matrix1,matrix2,rows,col);
    display_matrices(summation_matrix,rows,col);
    return 0;
}

File: Day_19/Q73.java
import java.util.*;
class Q73{
    static Scanner sc = new Scanner(System.in);
    //Input The Matrix
    static void input_2D_array(int arr[][],int row,int col){
        for(int i = 0;i < row;i++){
            for(int j = 0;j < col;j++) arr[i][j] = sc.nextInt();
        }
    }
    //Display The Matrix
    static void display_2D_array(int arr[][],int row,int col){
        for(int i = 0;i < row;i++){
            for(int j = 0;j < col;j++) System.out.print(arr[i][j] + "\t");
            System.out.println();
        }
    }
    //Matrix Addition
    static int[][] matrix_addition(int matrix1[][],int matrix2[][],int row,int col){
        int summation_matrix[][] = new int[row][col];
        for(int i = 0;i < row;i++){
            for(int j = 0;j < col;j++){
                summation_matrix[i][j] = matrix1[i][j] + matrix2[i][j];
            }
        }
        return summation_matrix;
    }
    public static void main(String[] args){
        int row = sc.nextInt();
        int col = sc.nextInt();
        //Input Validation
        if(row <= 0 || col <= 0){
            System.out.print("Invalid Input");
        }
        else{
            int matrix1[][] = new int[row][col];
            int matrix2[][] = new int[row][col];
            int summation[][] = new int[row][col];
            input_2D_array(matrix1, row, col);
            input_2D_array(matrix2, row, col);
            summation = matrix_addition(matrix1, matrix2, row, col);
            display_2D_array(summation, row, col);
        }
    }
}